src/modules/cluster/cluster-list/cluster-list.js

```diff
@@ -0,0 +1,84 @@
+(function() {
+    "use strict";
+
+    var app = angular.module("TendrlModule");
+
+    app.controller("clusterController", clusterController);
+
+    /*@ngInject*/
+    function clusterController($scope, $state, $interval, config, $filter, utils) {
+        var vm = this,
+            key,
+            len,
+            temp = [],
+            clusterData,
+            cluster,
+            i;
+
+        vm.importCluster = importCluster;
+        vm.getOrderByArg = getOrderByArg;
+        vm.displayArg = displayArg;
+
+        init();
+
+        function init() {
+            _createClusterList();
+        }
+
+        /*Refreshing list after each 30 second interval*/
+        var timer = $interval(function () {
+            
+            utils.getClusterData()
+                .then(function(data) {
+                    config.clusterData = data;
+                });
+            
+            init();
+
+        }, 1000 * config.initData.refreshIntervalTime );
+
+        /*Cancelling interval when scope is destroy*/
+        $scope.$on('$destroy', function() {
+            $interval.cancel(timer);
+        });
+
+        function displayArg() {
+            console.log(vm.orderByArg);
+        }
+
+        function importCluster() {
+            $state.go("import-cluster");
+        }
+
+        function getOrderByArg() {
```

Comment:
Can we do all sorting and filtering in html page only . in that case this "getOrderByArg" is not required